tzcodegen/stubs/authentic_execution.c

```
@@ -637,31 +610,24 @@ TEE_Result handle_input(void *session, uint32_t param_types, TEE_Param params[4]
 			tag_spongent[n] = ((uint8_t *)params[3].memref.buffer)[n];
 		}
 
-		printf("\n");
-		TEE_GetREETime(&t);
-		printf("time before SPONGENT decryption in handle_input(button_driver): %u.%03u\n", 
-							(unsigned int)t.seconds, (unsigned int)t.millis);
 		SpongentUnwrap(connection->connection_key, aad, 16, encrypted_data,
 															size * 8, data, tag_spongent);
 
-		printf("\n");
-		TEE_GetREETime(&t);
-		printf("time after SPONGENT decryption in handle_input(button_driver): %u.%03u\n", 
-							(unsigned int)t.seconds, (unsigned int)t.millis);
 		
 		for(int i = 0; i < size; i++){
-			printf("%2X", data[i]);
+			DMSG("%2X", data[i]);
 	  	}
-		printf("\n");
 
 	}// if spongent
 
 	connection->nonce = connection->nonce + 1;
 	params[0].value.a = 0;
 	params[0].value.b = 0;
-	printf("Calling input with ID %d\n", connection->io_id);
-    input_funcs[connection->io_id](session, param_types, params, data, size);
 	
+	if(connection->io_id >= 0 && connection->io_id < NUM_INPUTS) {
+		input_funcs[connection->io_id](session, param_types, params, data, size);
+	}
```

maybe return an error in the `else`? Same in `handle_entry`